store/multi_writer_app_store.go

```diff
@@ -262,12 +264,30 @@ func (s *MultiWriterAppStore) Prune() error {
 }
 
 func (s *MultiWriterAppStore) GetSnapshot() Snapshot {
+	snapshot, err := s.GetSnapshotAt(0)
+	if err != nil {
+		panic(err)
+	}
+	return snapshot
+}
+
+func (s *MultiWriterAppStore) GetSnapshotAt(version int64) (Snapshot, error) {
 	defer func(begin time.Time) {
 		getSnapshotDuration.Observe(time.Since(begin).Seconds())
 	}(time.Now())
-	appStoreTree := (*iavl.ImmutableTree)(atomic.LoadPointer(&s.lastSavedTree))
+
+	var err error
+	var appStoreTree *iavl.ImmutableTree
+	if version == 0 {
+		appStoreTree = (*iavl.ImmutableTree)(atomic.LoadPointer(&s.lastSavedTree))
+	} else {
+		appStoreTree, err = s.appStore.tree.GetImmutable(version)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to load immutable tree for version %v", version)
```

Comment:
Should we panic instead of returning the error? It would simplify the code quite a bit.

Reply:
It makes sense for `GetSnapshot` to panic as there should always be a lastest version. 
However, there are legitimate reasons for there not to be data available for an arbitrary version, as we can prune versions. 
In Application.ReplayApplication https://github.com/loomnetwork/loomchain/blob/ethdebug/app.go#L950 we take account of this.